rasa/core/training/generator.py

```diff
@@ -192,7 +220,22 @@ def _phase_name(everything_reachable_is_reached, phase):
         else:
             return f"data generation round {phase}"
 
+    def _generate_ml_trackers(self) -> List[TrackerWithCachedStates]:
+        steps = [step for step in self.story_graph.ordered_steps() if not step.is_rule]
+
+        return self._generate(steps, is_rule_data=False)
+
+    def _generate_rule_trackers(self) -> List[TrackerWithCachedStates]:
+        steps = [step for step in self.story_graph.ordered_steps() if step.is_rule]
+
+        return self._generate(steps, is_rule_data=True)
```

Comment:
does it makes sense to create 2 generators in agent, rather than making 1 generator handle both rules and ml?